### PR TITLE
feat(deployment): execute commands on device deployments

### PR DESCRIPTION
**Pull request by RomilShah**
[demo](https://asciinema.org/a/xRMD6MXugenrDBTesvX5sMh82)

### CODE REVIEW

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:71]: todo: use find_device_by name from `device/util.py`.

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:55]: todo: Update docstring
```
Execute commands on device deployments.
```

File: riocli/utils/execute.py
Comment (pallabpain) [inline riocli/utils/execute.py:45]: todo: Move the exec selection to the `rio deployment execute` command. Make this method throw an error when `exec_name` is `None` when the command is supposed to run on a deployment.

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:68]: todo: If no `exec_name` is given and the deployment has only one executable, use that executable in the `run_on_device` command. If the deployment has multiple executables and no `exec_name` is given, then `show_selection` and let the user select one.

You can move the logic you added in `run_on_device` here.

**Review by pallabpain — CHANGES_REQUESTED** [review]

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:69]: Move this line inside `exec_name is None` check. You don't want to fetch if not needed.

File: riocli/utils/execute.py
Comment (pallabpain) [inline riocli/utils/execute.py:40]: This is not needed here anymore.

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:80]: question: What will happen if the user provides an incorrect executable name?

If the error returned is not as intuitive, then it makes sense to also validate the user provided exec name against the ones in the package.

**Review by pallabpain — CHANGES_REQUESTED** [review]

File: riocli/deployment/execute.py
Comment (RomilShah) [reply riocli/deployment/execute.py:69]: now we will need it outside to verify if the user package contains exec_name provided by user.
In case user hasn't provided exec_name it will be needed to populate the exec_name list.

File: riocli/deployment/execute.py
Comment (RomilShah) [reply riocli/deployment/execute.py]: error seems to be intuitive to me.

```
python3 -m riocli deployment execute testngindep   curl localhost  --exec romiq
Usage:
  dectl exec EXECUTABLE_NAME COMMAND [ARG...] [flags]

Flags:
  -h, --help   help for exec

Global Flags:
      --config string                   (default "/opt/rapyuta/device_edge/deviceedge.yaml")
      --log-flush-frequency duration   Maximum number of seconds between log flushes (default 5s)

2024-08-15 19:51:30.552806 I | finding executable: executable not found
```

File: riocli/device/util.py
Comment (pallabpain) [inline riocli/device/util.py:89]: @RomilShah Please don't change this. The device manager API now supports filtering device list by name. This code was updated to save on iterating over the list.

What issue did you face that you had to revert this?

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:94]: Don't need color for this output.
```suggestion
        click.echo(response)
```

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:66]:
```suggestion
            click.secho(f'Only device runtime is supported.', fg=Colors.RED)
```

File: riocli/deployment/execute.py
Comment (pallabpain) [inline riocli/deployment/execute.py:78]: thought: This call is becoming expensive since we are invoking two device APIs,
1. To find the device by name
2. Again in `run_on_device`.


We should just move the logic to `run_on_device` such that if device_name is given, it invokes `find_device_by_name`.

**Review by pallabpain — CHANGES_REQUESTED** [review]

File: riocli/utils/execute.py
Comment (pallabpain) [inline riocli/utils/execute.py:26]: question: This helper is used in other places as well. Can you verify whether they are affected or not?

File: riocli/device/execute.py
Comment (pallabpain) [inline riocli/device/execute.py:32]: question: Why do this? We don't need to modify this command.

File: riocli/utils/execute.py
Comment (pallabpain) [inline riocli/utils/execute.py:41]: Why not invoke `find_device_by_name` helper helper here? It does the same thing that you are doing here.

**Review by pallabpain — CHANGES_REQUESTED** [review]

File: riocli/utils/execute.py
Comment (RomilShah) [reply riocli/utils/execute.py:26]: Yes, none of them are affected since all the invoke calls are parameterized

File: riocli/device/execute.py
Comment (RomilShah) [reply riocli/device/execute.py:32]: I thought of adding the functionality of running commands on deployment too.
but now that I think, its not needed, hence removing it

File: riocli/utils/execute.py
Comment (RomilShah) [reply riocli/utils/execute.py:41]: it was causing circular import issue

**Review by pallabpain — APPROVED** [review]
Approving the PR. But don't merge it yet. I want to confirm a few changes before merging this with base branch.

**Pull request merged**
